specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,985 @@
+  br label %continue__1.i.i.i
+
+continue:
+...
+```
+
+Similarly to **Bullet 9**, the phi instruction can be used to move classical floating point values between branches.
+
+**Bullet 11: Classical fixed point computations computations** <br />
+If a backend has support for fixed point operations, they can also use the following intrinsics:
+Additionally if an adaptive profile program has support for floating point computations, the following instructions are supported:
+| LLVM Intrinsic | Context and Purpose               | Note                        |
+|:-----------------|:----------------------------------|:----------------------------|
+| `llvm.smul.fix.*`| Used to multiply two signed fixed point numbers.  |                             |
+| `llvm.smul.fix.sat*`| Same as above but clamps to min/max number in scale.  |                             |
+| `llvm.umul.fix.*`| Used to multiply two unsigned fixed point numbers. |                             |
+| `llvm.umul.fix.sat*`| Same as above but clamps to min/max number in scale.  |                             |
+| `llvm.sdiv.fix.*`| Used to divide two signed fixed point numbers. | Can cause real-time errors.                             |
+| `llvm.sdiv.fix.sat*`| Same as above but clamps to min/max number in scale.  |                             |
+| `llvm.udiv.fix.*`| Used to divide two unsigned fixed point numbers. | Can cause real-time errors. |
+| `llvm.udiv.fix.sat*`| Same as above but clamps to min/max number in scale.  |                             |
+|                  |                                   |                             |
+
+Similarly to **Bullet 9**, the phi instruction can be used to move classical fixed point values between branches.
+
+
+**Bullet 12: User defined functions and function calls** <br />
+An adaptive profile program may use user defined functions and function calls.
+For example, consider that with user defined functions if a back-end has a `Cnot` gate in its instruction set,
+then a program that liberally uses `Swap` operations can define a function and call it as follows:
+```llvm
+define void @swap(%Qubit* %arg1, %Qubit* %arg2) {
+call void __quantum__qis__cnot__body(%Qubit* %arg1, %Qubit* %arg2)
+call void __quantum__qis__cnot__body(%Qubit* %arg2, %Qubit* %arg1)
+call void __quantum__qis__cnot__body(%Qubit* %arg1, %Qubit* %arg2)
+}
+
+define void @main() {
+...
+call void @swap_0_1()
+...
+}
+```
+
+Moreover, classical functions can be defined in the IR assuming that a back-end has opted into supporting
+classical computations:
+
+```llvm
+define i64 @triple(i64 %0) {
+%1 = mul i64 %0, 3
+ret i64 %1
+}
+
+define void @main() {
+...
+%0 = call void @triple(i64 2)
+...
+}
+```
+
+The only restriction on user functions and definitions is that you cannot have dynamically allocated `%Qubit*`  arguments, they must still be constant `%Qubit*` id's. 
+
+**Bullet 13: Dynamically computed or linked float angles for gates** <br />
```

Comment:
Based on our conversation, this option could be called "Non-static float angles for gates" or something similar.